### PR TITLE
Add Rust Hello World

### LINKED ISSUE #1307: hello world Rust library
in an effort to explore #1306, let's start a Rust apache module.

I think writing our own proxy may a solid option. Many proxies have lots of extra features for lots of use cases. Our use case is specific and I'd like to keep the attack/configuration surface extremely small.

I think Rust is a really strong language choice for such a use case given it's safety, speed and ease of deployment.

So this ticket is fairly simple, just to get us started exploring this option.

* [x] `rake build:mod-ood-proxy` task
* [x] the Rake task builds a binary that prints 'hello world'

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201735133575781/1201737325493267) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by msquee**
This PR adds a Rake task for building and running the Hello World Rust binary.

To build the binary inside of a Docker container:
`rake ood_proxy:build`

Start hello world:
`rake ood_proxy:start`

```sh
$ rake ood_proxy:build
docker build --build-arg VERSION=v2.0.13-2d0b2e4 -t ood-proxy-rs:2.0.13-2d0b2e4 -f Dockerfile.proxy .
Sending build context to Docker daemon  27.01MB
Step 1/9 : FROM rust:latest as builder
 ---> ceae1d63f79e
Step 2/9 : WORKDIR /app
 ---> Using cache
 ---> 998389cfbcca
Step 3/9 : COPY ood-proxy-rs/ .
 ---> Using cache
 ---> cc29fa407cc6
Step 4/9 : RUN cargo build --release
 ---> Using cache
 ---> 1d72dd1fb68e
Step 5/9 : FROM debian:buster-slim as runtime
 ---> df0140a4030c
Step 6/9 : WORKDIR /app
 ---> Using cache
 ---> 4f748015c063
Step 7/9 : RUN apt-get update -y     && apt-get install -y --no-install-recommends openssl     && apt-get autoremove -y     && apt-get clean -y     && rm -rf /var/lib/apt/lists/*
 ---> Using cache
 ---> ed9e9a50d63e
Step 8/9 : COPY --from=builder /app/target/release/ood-proxy-rs ood-proxy-rs
 ---> Using cache
 ---> cc040b441483
Step 9/9 : ENTRYPOINT ["./ood-proxy-rs"]
 ---> Using cache
 ---> e9ac73616865
[Warning] One or more build-args [VERSION] were not consumed
Successfully built e9ac73616865
Successfully tagged ood-proxy-rs:2.0.13-2d0b2e4
docker tag ood-proxy-rs:2.0.13-2d0b2e4 ood-proxy-rs:latest
gitpod /workspace/ondemand $ rake ood_proxy:start
docker run ood-proxy-rs --name ood-proxy-rs ood-proxy-rs:latest
Hello Open OnDemand!
gitpod /workspace/ondemand $
```

Closes #1307

### CODE REVIEW

**Comment by johrstrom** [comment]
We don't want any of the runtime stuff. Just build. Right now I just want to be able to build a `.so` file. At some point we'll then COPY that into the main `ood` (or more likely `ood-test/dev` image).

**Comment by johrstrom** [comment]
Oh also, change the task to `rake build:proxy`. Putting this on a feature branch so we can do that, but I want a cohesive rake task structure like:

```
build:
   this
   that
   the other
package:
   this
test:
   a 4th thing
```

File: lib/tasks/ood-proxy.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    'ood-proxy-rs' || ENV['OOD_PROXY_CONTAINER_NAME'].to_s
+  end
+
+  task :proxy_docker do
+    sh build_cmd("Dockerfile.proxy", proxy_container_name)
+    sh tag_latest_container_cmd(proxy_container_name)
+  end
+
+  task :proxy do
+    build_args = ['cargo', 'build']
```
Comment (treydock) [inline lib/tasks/ood-proxy.rb:17]: Just wanted to point out that having a direct build Rake task will work very well with the RPM build.  For EL7 builds the `cargo` package comes from EPEL and for EL8 builds it comes from the OS.

File: Rakefile [hunk 2]
```diff
@@ -52,6 +54,10 @@ class Component
     @path.join('app.js').exist?
   end
 
+  def rust_app?
```
Comment (treydock) [inline Rakefile:57]: There needs to be something added to `build` namespace in this main Rakefile that will also build the ood-proxy-rs app.  This way the existing `rake build` inside the RPM spec will pick up this new app.  

One idea I can imagine is changing the `infrastructure` function from array generating components to hash where key is app name like in the array and value is the name of the build task, then pass that extra info into Component and reference the build task name somewhere in build namespace after looping over infrastructure apps.

**Comment by johrstrom** [comment]
Let's merge this now and just change it later.

**Pull request merged**